odxtools/diaglayer.py

```diff
@@ -529,11 +533,12 @@ def get_can_func_req_id(self, protocol_name: Optional[str] = None) \
 
         return int(result)
 
-    def get_doip_logical_gateway_address(self, protocol_name: Optional[str] = None) \
+    def get_doip_logical_gateway_address(self, functional: Optional[bool] = False, protocol_name: Optional[str] = None) \
```

Comment:
There is a lot of duplicate code, which I think is hard to maintain. I therefore propose the following change: We write a generic method `def get_doip_address(self, address_type : DoIPAddressType, is_functional : bool, protocol_name: str)`, where DoIPAddressType is a Enum with values `LogicalGatewayAddress, LogicalTesterAddress, LogicalEcuAddress, LogicalFunctionalAddress`.

Reply:
I absolutely agree that invoking such a function with an extra parameter is a bit too verbose. Using helper functions is probably the way to go. I will however, address this in a future PR where I plan to introduce more `get_comparam_XY` functions 😆